driver/razerkbd_driver.c

```diff
@@ -3879,13 +3879,26 @@ static int razer_kbd_probe(struct hid_device *hdev, const struct hid_device_id *
     struct usb_interface *intf = to_usb_interface(hdev->dev.parent);
     struct usb_device *usb_dev = interface_to_usbdev(intf);
     struct razer_kbd_device *dev = NULL;
+    struct razer_kbd_usb_device_data *usb_dev_data = NULL;
 
     dev = kzalloc(sizeof(struct razer_kbd_device), GFP_KERNEL);
     if(dev == NULL) {
         dev_err(&intf->dev, "out of memory\n");
         return -ENOMEM;
     }
 
+    // Allocate data in context to the usb device if not already done.
+    if (dev_get_drvdata(&usb_dev->dev) == NULL) {
```

Comment:
How about this? Then we also have `usb_dev_data` set to a value when drvdata has been set already
```c
usb_dev_data = dev_get_drvdata(&usb_dev->dev);
if (!usb_dev_data) {
```